### PR TITLE
Proved postulates on System model

### PR DESCRIPTION
**Pull request by lisandrasilva**
Proved the following postulates on the `LibraBFT.Yasm.System`:
- `cheatStepDNMPeerStates`
- `msgs-stable`
- `sendMessages-target`
- `peersRemainInitialized`

In order to prove the last one I needed to add a new parameter to the `SystemParameters` which is a `Decidable` over `PeerId`. Actually, I think that this new parameter will also be useful for some other lemmas that we may need to prove in future.

### CODE REVIEW

File: LibraBFT/Yasm/Base.agda
Comment (mark-moir) [inline LibraBFT/Yasm/Base.agda:49]: Funnily enough, just yesterday I did the same in the branch I am working on.  Could you move this to be the second parameter (right after `PeerId`), which will avoid a conflict with my similar change.  Also, please spell it like this:
```
  _≟PeerId_ : (p1 p2 : PeerId) → Dec (p1 ≡ p2)
```

File: LibraBFT/Yasm/System.agda
Comment (mark-moir) [inline LibraBFT/Yasm/System.agda:259]: Please remove the extraneous space before the second parameter.

**Review by mark-moir — CHANGES_REQUESTED** [review]
Thanks @lisandrasilva, looks good.  Before I can merge, I need you to send one more commit with signoff.  I have made some super minor suggestions to include in the commit.  Thanks!

File: LibraBFT/Yasm/Base.agda
Comment (lisandrasilva) [reply LibraBFT/Yasm/Base.agda]: Done!

**Review by mark-moir — APPROVED** [review]
Looks good.   Thanks @lisandrasilva.

**Pull request merged**
